lib/render/px_vp20/utils.h

```diff
@@ -115,6 +117,40 @@ class px_vp20Utils
 };
 
 
+/// Simple RAII class to save uniform buffer bindings, to deal with a Maya
+/// issue.
+///
+/// XXX: When Maya is using OpenGL Core Profile as the rendering engine (in
+/// either compatibility or strict mode), batch renders like those done in the
+/// "Render View" window or through the ogsRender command do not properly track
+/// uniform buffer binding state. This was causing issues where the first batch
+/// render performed would look correct, but then all subsequent renders done
+/// in that Maya session would be completely black (no alpha), even if the
+/// frame contained only Maya-native geometry or if a new scene was
+/// created/opened.
+///
+/// To avoid this problem, this object can be used to save and restore Maya's
+/// uniform buffer bindings across Hydra/OpenGL calls. We try not to bog down
+/// performance by saving and restoring *all* GL_MAX_UNIFORM_BUFFER_BINDINGS
+/// possible bindings, so instead we only do just enough to avoid issues.
+/// Empirically, the problematic binding has been the material binding at
+/// index 4.
+class GLUniformBufferBindingsSaver
+{
+    public:
+        static constexpr size_t UNIFORM_BINDINGS_TO_SAVE = 5u;
```

Comment:
This looks like an implementation detail and should be hidden in the private section?

Reply:
Yup, makes sense! Done in commit 5c448b5bdb5e23bfe3aaba7a230c49cadc723e74.